Add drop feedback UX

mjbvz opened the pull request:
For #179430

Adds two new UX components:

- An inline progress icon shown when a drop operation takes over 500ms. This replaces the notification. You can click on it to cancel the drop

- Post drop, a drop feedback icon that lets you drop the file in a different way. This lets you drop the file as plain text for instance instead of as a markdown link



https://user-images.githubusercontent.com/12821956/230517919-c4c075c2-3455-49d5-aa98-412f6233e5b5.mov

gjsjohnmurray commented:
Looks good. Did you try positioning the icon at the beginning of what gets dropped rather than at the end? Would that require less mouse movement?

mjbvz commented:
@gjsjohnmurray Thanks for the feedback. I went with the end because this matches what word and outlook do when pasting. However I've updated the widget so it should now always appear on screen, even if the end position is actually off screen. This makes it easier to reach

I've also updated the widget to dismiss as soon as you move the cursor somewhere out of the main edit range

joyceerhl commented on src/vs/editor/contrib/dropIntoEditor/browser/dropProgressWidget.ts:
```diff
@@ -0,0 +1,147 @@
+export class DropProgressManager extends Disposable {
+
+	/** Delay before showing the progress widget */
+	private readonly _showDelay = 500; // ms
```
I think the equivalent default delay is 150ms in notification progress, any reason to deviate? https://github.com/mjbvz/vscode/blob/b45c925dd0296392abf6ab3911c207fa32924b1d/src/vs/workbench/services/progress/browser/progressService.ts#L82

mjbvz replied:
This widget shifts around the line text so we want to be cautious about showing it too quickly. That's why I went with half a second instead
